custom_components/sector/descriptions.py

```diff
@@ -0,0 +1,13 @@
+# descriptions.py
+from dataclasses import dataclass
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.helpers.entity import EntityDescription
+
+@dataclass
+class GenericSensorEntityDescription(EntityDescription):
+    """A generic entity description for multiple types of sensors."""
+    device_class: str | None = None
+    unit_of_measurement: str | None = None
+    state_class: str | None = None
+    icon: str | None = None
+    # Additional properties for flexibility
```

Comment:
These are not needed as they are `None` by default e.g. if they don't exist, it will return `None`

Reply:
Moved to const.py

Reply:
Removed, utilizing BinarySensorEntityDescription in binary_sensor.py instead.